streams/src/main/java/org/apache/kafka/streams/state/internals/QueryableStoreProvider.java

```diff
@@ -45,24 +43,28 @@ public QueryableStoreProvider(final List<StateStoreProvider> storeProviders,
      *
      * @param storeName          name of the store
      * @param queryableStoreType accept stores passing {@link QueryableStoreType#accepts(StateStore)}
+     * @param includeStaleStores     if true, include standbys and recovering stores;
+     *                                        if false, only include running actives.
      * @param <T>                The expected type of the returned store
      * @return A composite object that wraps the store instances.
      */
     public <T> T getStore(final String storeName,
-                          final QueryableStoreType<T> queryableStoreType) {
+                          final QueryableStoreType<T> queryableStoreType,
+                          final boolean includeStaleStores) {
         final List<T> globalStore = globalStoreProvider.stores(storeName, queryableStoreType);
         if (!globalStore.isEmpty()) {
-            return queryableStoreType.create(new WrappingStoreProvider(singletonList(globalStoreProvider)), storeName);
+            return queryableStoreType.create(globalStoreProvider, storeName);
```

Comment:
thinking aloud: guess there is nt much value in wrapping a single provider.. so +1

Reply:
Yeah, the role of the wrapping store provider really is nothing more than just iterating over the list of providers. For global stores, there's always exactly one provider, so it's purely a performance penalty for no gain at all.